editor.planx.uk/src/pages/FlowEditor/components/Settings/Submissions/components/FormattedResponse.tsx

```diff
@@ -14,20 +16,23 @@ const Response = styled(Box)(({ theme }) => ({
   backgroundColor: theme.palette.background.paper,
 }));
 
+const InvalidJSONError = () => <Typography>Error parsing response</Typography>
+
 export const FormattedResponse: React.FC<Submission> = (submission) => {
-  return submission.eventType === "Pay" ? (
-    <Response component="pre">
-      {JSON.stringify(submission.response, null, 2)}
-    </Response>
-  ) : (
+  const getResponse = ({ eventType, status, response }: Submission) => {
```

Comment:
Consolidated this logic to a single function instead of splitting it over multiple terneries within the template.